File: Server/src/main/java/com/inther/dto/MessageDto.java
package com.inther.dto;

import com.inther.assets.validators.RequestDataValidator;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import javax.validation.constraints.*;
import java.io.Serializable;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class MessageDto implements Serializable
{
    @Null(groups = {RequestDataValidator.AddMessage.class})
    @NotNull(groups = {RequestDataValidator.UpdateMessage.class, RequestDataValidator.DeleteMessage.class})
    private UUID id;

    @NotNull(groups = {RequestDataValidator.AddMessage.class})
    private UUID presentationId;

    @NotNull(groups = {RequestDataValidator.AddMessage.class})
    private UUID userId;

    @Email
    @NotBlank(groups = {RequestDataValidator.AddMessage.class})
    private String email;

    @NotBlank(groups = {RequestDataValidator.AddMessage.class})
    private String text;

    @NotBlank(groups = {RequestDataValidator.AddMessage.class})
    private String type;

    @NotNull(groups = {RequestDataValidator.AddMessage.class})
    private Boolean anonymous;
}

File: Client/src/app/models/dtos/message.dto.ts
import { Message } from '../message.model';

export class MessageDTO {
  id: string;
  presentationId: string;
  userId: string;
  email: string;
  text: string;
  type: string;
  anonymous: boolean;

  constructor(obj: Object = {}) {
    Object.assign(this, obj);
  }

  static toModel(m: MessageDTO): Message {
    return <Message>{
      id: m.id,
      presentationId: m.presentationId,
      userId: m.userId,
      email: m.email,
      text: m.text,
      type: m.type,
      anonymous: m.anonymous
    };
  }
}
